spinalcordtoolbox/scripts/sct_process_segmentation.py

```diff
@@ -422,37 +439,60 @@ def main(argv: Sequence[str]):
         if key == 'length':
             # For computing cord length, slice-wise length needs to be summed across slices
             metrics_agg[key] = aggregate_per_slice_or_level(metrics[key], slices=slices,
                                                             levels=levels,
                                                             distance_pmj=distance_pmj, perslice=perslice,
                                                             perlevel=perlevel, fname_vert_level=fname_vert_level,
                                                             group_funcs=(('SUM', func_sum),), length_pmj=length_from_pmj)
+            if arguments.normalize == 'PAM50':
+                metrics_agg_PAM50_space[key] = aggregate_per_slice_or_level(metrics_PAM50_space[key], slices=slices,
+                                                                            levels=levels,
+                                                                            distance_pmj=distance_pmj, perslice=perslice,
+                                                                            perlevel=perlevel, fname_vert_level=fname_vert_level_PAM50,
+                                                                            group_funcs=(('SUM', func_sum),), length_pmj=length_from_pmj)
+
         else:
             # For other metrics, we compute the average and standard deviation across slices
             metrics_agg[key] = aggregate_per_slice_or_level(metrics[key], slices=slices,
                                                             levels=levels,
                                                             distance_pmj=distance_pmj, perslice=perslice,
                                                             perlevel=perlevel, fname_vert_level=fname_vert_level,
                                                             group_funcs=group_funcs, length_pmj=length_from_pmj)
+            if arguments.normalize == 'PAM50':
+                metrics_agg_PAM50_space[key] = aggregate_per_slice_or_level(metrics_PAM50_space[key], slices=slices,
+                                                                            levels=levels,
+                                                                            distance_pmj=distance_pmj, perslice=perslice,
+                                                                            perlevel=perlevel, fname_vert_level=fname_vert_level_PAM50,
+                                                                            group_funcs=group_funcs, length_pmj=length_from_pmj)
+
     metrics_agg_merged = merge_dict(metrics_agg)
+    # Save metrics in PAM50 anatomical dimensions
+    if arguments.normalize == 'PAM50':
+        metrics_agg_merged_PAM50_space = merge_dict(metrics_agg_PAM50_space)
+        file_out_norm_PAM50 = add_suffix(file_out, '_PAM50')
+        save_as_csv(metrics_agg_merged_PAM50_space, file_out_norm_PAM50, fname_in=fname_segmentation, append=append)
```

Comment:
yes we need both files, afterwards for when computing the MSCC script : see  PR #4003, I am not against having to run it twice, but it will still be recomputing the same morphometrics on the segmentation before adjusting with PAM50

Reply:
That's a good point -- We would be recomputing the same `metrics` object for both calls:

<details>

Without `-normalize PAM50`:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/72b6317c65272292a563e3b8778b2e7f80dfe448/spinalcordtoolbox/scripts/sct_process_segmentation.py#L419-L423

With `-normalize PAM50`:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/72b6317c65272292a563e3b8778b2e7f80dfe448/spinalcordtoolbox/scripts/sct_process_segmentation.py#L419-L428

</details>

That said, we kind of already repeat morphometrics computations? For example, in the ["Computing metrics"](https://spinalcordtoolbox.com/user_section/tutorials/registration-to-template/shape-metric-computation.html) section of the tutorials:

![image](https://user-images.githubusercontent.com/16181459/214654356-98054295-3c6b-4c2d-ad42-de7774ac6a9e.png)

For several of these commands, they perform the same morphometric computation -- the only difference is the aggregation.

Reply:
One other factor is the idea that "1 call to `sct_process_segmentation` == 1 output csv file". I think that feels intuitive to me? And I think there's a benefit to keeping that pattern, even if it means repeating some of the computation.

Reply:
Done in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3977/commits/41c9b6dc3778b0c85cdc0893f2d5e3d641ae13e3